### PR TITLE
Assignment cleanup

### PR DESCRIPTION
**Pull request by denisbsu**

### CODE REVIEW

File: crates/network-scheduler/src/storage.rs [hunk 1]
```diff
@@ -358,6 +358,49 @@ impl S3Storage {
...
+            .await
+        else {
+            log::error!("Failed to list assignments");
+            return;
+        };
+        prometheus_metrics::s3_request();
+        for v in files
+            .contents()
+            .iter()
+            .skip(Config::get().assignment_history_len)
```
Comment (kalabukdima) [inline crates/network-scheduler/src/storage.rs:378]: Wouldn't it remove everything except for the first N files? It should probably be the reverse

Comment (denisbsu) [reply crates/network-scheduler/src/storage.rs:378]: fixed

File: crates/network-scheduler/src/storage.rs [hunk 2]
```diff
@@ -358,6 +358,50 @@ impl S3Storage {
...
+                .delete_object()
+                .bucket(&self.config.scheduler_state_bucket)
+                .key(filepath)
+                .send()
+                .await;
+            prometheus_metrics::s3_request();
+            match result {
+                Ok(result) => log::error!("S3 object deleted: {result:?}"),
+                Err(err) => {
+                    log::error!("Can not delete S3 object: {err:?}");
```
Comment (kalabukdima) [inline crates/network-scheduler/src/storage.rs]: I would also mention which object it's trying to delete. I may be difficult to get the context in the mess of scheduler logs

File: crates/network-scheduler/src/storage.rs [hunk 3]
```diff
@@ -358,6 +358,50 @@ impl S3Storage {
...
+            let result = self
+                .client
+                .delete_object()
+                .bucket(&self.config.scheduler_state_bucket)
+                .key(filepath)
+                .send()
+                .await;
+            prometheus_metrics::s3_request();
+            match result {
+                Ok(result) => log::error!("S3 object deleted: {result:?}"),
```
Comment (kalabukdima) [inline crates/network-scheduler/src/storage.rs]: Why is it an error?

File: crates/network-scheduler/src/cli.rs [hunk 4]
```diff
@@ -16,6 +16,10 @@ fn default_worker_version() -> VersionReq {
     ">=1.0.0-rc3".parse().unwrap()
 }
 
+fn default_assignment_history_len() -> usize {
+    10
```
Comment (kalabukdima) [inline crates/network-scheduler/src/cli.rs]: Let's make it 1000 by default

**Review by kalabukdima — APPROVED** [review]

File: crates/network-scheduler/src/cli.rs (hunk 4 above)
Comment (denisbsu) [reply crates/network-scheduler/src/cli.rs]: ok

File: crates/network-scheduler/src/storage.rs (hunk 3 above)
Comment (denisbsu) [reply crates/network-scheduler/src/storage.rs]: fair enough, demote to debug

File: crates/network-scheduler/src/storage.rs (hunk 2 above)
Comment (denisbsu) [reply crates/network-scheduler/src/storage.rs]: done

**Pull request merged**